### PR TITLE
nanodbc: Bump dependency: odbc/2.3.9

### PR DESCRIPTION
**Pull request by alejandro-colomar**
Specify library name and version:  **nanodbc/all**

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by madebr — APPROVED** [review]

**Review by Croydon — APPROVED** [review]

**Review by ericriff — APPROVED** [review]

File: recipes/nanodbc/all/conanfile.py [hunk 1]
```diff
@@ -62,7 +62,7 @@ def requirements(self):
         if self.options.with_boost:
             self.requires("boost/1.73.0")
```
Comment (ericriff) [inline recipes/nanodbc/all/conanfile.py:63]: Any reason to keep the outdated boost?

**Review by SSE4 — APPROVED** [review]

File: recipes/nanodbc/all/conanfile.py (hunk 1 above)
Comment (SSE4) [reply recipes/nanodbc/all/conanfile.py:63]: no idea, but probably out of the scope of this PR

Comment (alejandro-colomar) [reply recipes/nanodbc/all/conanfile.py:63]: I didn't notice.  I can add it to this PR

Comment (alejandro-colomar) [reply recipes/nanodbc/all/conanfile.py:63]: Too late :)

**Pull request merged**